### PR TITLE
Ack SETTINGS frame immediately

### PR DESCRIPTION
**Pull request by nmittler**
This is a partial fix for #243.

### CODE REVIEW

**Comment by nmittler** [comment]
@dsymonds Can you take a look?

File: transport/http2_client.go [hunk 1]
```diff
@@ -590,9 +591,10 @@ func (t *http2Client) handleSettings(f *http2.SettingsFrame) {
 				}
 			case http2.SettingInitialWindowSize:
 				t.mu.Lock()
+				delta := int(v - t.streamSendQuota)
 				for _, s := range t.activeStreams {
 					// Adjust the sending quota for each s.
-					s.sendQuotaPool.reset(int(v - t.streamSendQuota))
+					s.sendQuotaPool.reset(delta)
```
Comment (dsymonds) [inline transport/http2_client.go]: revert this. Please try to minimise gratuitous changes.

**Comment by dsymonds** [comment]
I'm not really qualified to review this. @iamqizhao is the right person.

**Comment by iamqizhao** [comment]
I somehow lost the access to github because I am in China. Will be back
next week.

File: transport/http2_client.go (hunk 1 above)
Comment (nmittler) [inline transport/http2_client.go]: done.

**Comment by iamqizhao** [comment]
This PR is incorrect. The control frames have to be sent by the controller.  If you can wait, I will make one for you next Monday.

**Comment by nmittler** [comment]
@iamqizhao OK sounds good.  I'l just close this then ... #243 is already assigned to you.

**Pull request closed without merging**